机器人启动失败

### 是否会提供贡献？

否

### 版本信息

    implementation("love.forte.simbot:simbot-core:4.13.0")
    implementation("love.forte.simbot.component:simbot-component-kook-core:4.1.6")

### 涉及的编程语言

Kotlin

### 项目构建工具

Gradle

### 内容描述

`Exception in thread "main" kotlinx.serialization.json.internal.JsonDecodingException: Failed to parse literal 'true' as an int value at element: $.enable_open
JSON input: .....:"************","welcome_channel_id":"************"}
	at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:24)
	at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:32)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.unparsedPrimitive(TreeJsonDecoder.kt:99)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeTaggedInt(TreeJsonDecoder.kt:506)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeTaggedInt(TreeJsonDecoder.kt:37)
	at kotlinx.serialization.internal.TaggedDecoder.decodeInt(Tagged.kt:226)
	at love.forte.simbot.kook.util.BooleanToIntSerializer.deserialize(Serializers.kt:35)
	at love.forte.simbot.kook.util.BooleanToIntSerializer.deserialize(Serializers.kt:33)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:337)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableValue(Tagged.kt:207)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement$lambda$1(Tagged.kt:279)
	at kotlinx.serialization.internal.TaggedDecoder.tagBlock(Tagged.kt:294)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement(Tagged.kt:279)
	at love.forte.simbot.kook.objects.SimpleGuild$$serializer.deserialize(Guild.kt:87)
	at love.forte.simbot.kook.objects.SimpleGuild$$serializer.deserialize(Guild.kt:87)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:337)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableValue(Tagged.kt:207)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement$lambda$1(Tagged.kt:279)
	at kotlinx.serialization.internal.TaggedDecoder.tagBlock(Tagged.kt:294)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement(Tagged.kt:279)
	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:538)
	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(CollectionSerializers.kt:80)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:337)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableValue(Tagged.kt:207)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement$lambda$1(Tagged.kt:279)
	at kotlinx.serialization.internal.TaggedDecoder.tagBlock(Tagged.kt:294)
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement(Tagged.kt:279)
	at love.forte.simbot.kook.api.ListData$$serializer.deserialize(ApiResult.kt:137)
	at love.forte.simbot.kook.api.ListData$$serializer.deserialize(ApiResult.kt:137)
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:337)
	at kotlinx.serialization.json.internal.TreeJsonDecoderKt.readJson(TreeJsonDecoder.kt:26)
	at kotlinx.serialization.json.Json.decodeFromJsonElement(Json.kt:186)
	at love.forte.simbot.kook.api.ApiResult.parseData(ApiResult.kt:235)
	at love.forte.simbot.kook.api.ApiResult.parseDataOrThrow(ApiResult.kt:254)
	at love.forte.simbot.kook.api.ApiRequests__ApiRequestsKt.requestData(ApiRequests.kt:135)
	at love.forte.simbot.kook.api.ApiRequests.requestData(Unknown Source)
	at love.forte.simbot.kook.api.ApiRequests__ApiRequestsKt$requestData$1.invokeSuspend(ApiRequests.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:101)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:113)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:589)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:823)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:720)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:707)

FAILURE: Build failed with an exception.`

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,3 +1,23 @@
+/*
+ *     Copyright (c) 2025. ForteScarlet.
+ *
+ *     This file is part of simbot-component-kook.
+ *
+ *     simbot-component-kook is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     simbot-component-kook is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU Lesser General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with simbot-component-kook,
+ *     If not, see <https://www.gnu.org/licenses/>.
+ */
+
 import love.forte.gradle.common.core.repository.Repositories
 
 /*
@@ -47,7 +67,8 @@ apiValidation {
     ignoredProjects.addAll(
         listOf(
             "api-reader",
-            "message-element-processor"
+            "message-element-processor",
+            "internal-local-test"
         ),
     )
 

--- a/buildSrc/src/main/kotlin/P.kt
+++ b/buildSrc/src/main/kotlin/P.kt
@@ -34,7 +34,7 @@ object P : ProjectDetail() {
     override val homepage: String
         get() = HOMEPAGE
 
-    const val VERSION = "4.1.6"
+    const val VERSION = "4.1.7"
     const val NEXT_VERSION = "4.1.7"
 
     override val snapshotVersion = "$NEXT_VERSION-SNAPSHOT"

--- a/simbot-component-kook-api/src/commonMain/kotlin/love/forte/simbot/kook/util/Serializers.kt
+++ b/simbot-component-kook-api/src/commonMain/kotlin/love/forte/simbot/kook/util/Serializers.kt
@@ -1,18 +1,21 @@
 /*
- * Copyright (c) 2022-2023. ForteScarlet.
+ *     Copyright (c) 2022-2025. ForteScarlet.
  *
- * This file is part of simbot-component-kook.
+ *     This file is part of simbot-component-kook.
  *
- * simbot-component-kook is free software: you can redistribute it and/or modify it under the terms of
- * the GNU Lesser General Public License as published by the Free Software Foundation,
- * either version 3 of the License, or (at your option) any later version.
+ *     simbot-component-kook is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
  *
- * simbot-component-kook is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
- * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- * See the GNU Lesser General Public License for more details.
+ *     simbot-component-kook is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU Lesser General Public License for more details.
  *
- * You should have received a copy of the GNU Lesser General Public License along with simbot-component-kook,
- * If not, see <https://www.gnu.org/licenses/>.
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with simbot-component-kook,
+ *     If not, see <https://www.gnu.org/licenses/>.
  */
 
 package love.forte.simbot.kook.util
@@ -23,16 +26,36 @@ import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
 import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.encoding.Decoder
 import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
 
 
 /**
  * 将 [Boolean] 转为 [Int], 其中，false = 0, true = 1.
  *
  * 在反序列化的时候，0 = false，其他情况为true。
+ *
+ * since 4.1.7: 在 JSON 序列化下，对 Boolean 的转化会更宽松，支持同时解析 int/boolean/string 类型。
  */
 public object BooleanToIntSerializer : KSerializer<Boolean> {
     override fun deserialize(decoder: Decoder): Boolean {
-        return decoder.decodeInt() != 0
+        val jsonDecoder = decoder as? JsonDecoder
+            ?: return decoder.decodeInt() != 0
+
+        val decodeJsonElement = jsonDecoder.decodeJsonElement()
+        val primitive = decodeJsonElement as? JsonPrimitive
+            ?: error("Unexpected JSON element type: $decodeJsonElement")
+        if (primitive is JsonNull) {
+            return false
+        }
+
+        primitive.intOrNull?.let { return it != 0 }
+        primitive.booleanOrNull?.let { return it }
+
+        return when (primitive.content) {
+            "true", "TRUE", "True", "1" -> true
+            "false", "FALSE", "False", "0" -> false
+            else -> error("Unexpected JSON element type: $primitive")
+        }
     }
 
     override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("BooleanToInt", PrimitiveKind.BOOLEAN)

--- a/simbot-component-kook-core/src/commonTest/kotlin/love/forte/simbot/component/event/GuildEventsDeserializeTests.kt
+++ b/simbot-component-kook-core/src/commonTest/kotlin/love/forte/simbot/component/event/GuildEventsDeserializeTests.kt
@@ -5,7 +5,9 @@ import love.forte.simbot.annotations.ExperimentalSimbotAPI
 import love.forte.simbot.kook.Kook
 import love.forte.simbot.kook.event.*
 import kotlin.test.Test
+import kotlin.test.assertFalse
 import kotlin.test.assertIs
+import kotlin.test.assertTrue
 
 /**
  * Tests for KOOK guild events deserialization.
@@ -262,5 +264,92 @@ class GuildEventsDeserializeTests {
         val ser = Signal.Event.serializer(SystemExtra.serializer())
         val deserialized = json.decodeFromString(ser, rawJson)
         assertIs<UpdatedEmojiEventExtra>(deserialized.data.extra)
+    }
+
+    @OptIn(ExperimentalSimbotAPI::class)
+    @Test
+    fun testDeserializeUpdatedGuildEventWithBooleanEnableOpenTrue() {
+        //language=json
+        val rawJson =
+            """{
+  "s": 0,
+  "d": {
+    "channel_type": "GROUP",
+    "type": 255,
+    "target_id": "601630000000",
+    "author_id": "1",
+    "content": "[系统消息]",
+    "extra": {
+      "type": "updated_guild",
+      "body": {
+        "id": "601630000000",
+        "name": "test111",
+        "topic": "测试服务器主题",
+        "user_id": "2418xxx",
+        "icon": "https://xxx/icons/2020-05/YQyfHxxx.png/icon",
+        "notify_type": 1,
+        "region": "shanghai",
+        "enable_open": true,
+        "open_id": 1123123123,
+        "default_channel_id": "4881800000000",
+        "welcome_channel_id": "4881800000000"
+      }
+    },
+    "msg_id": "0108feaf-xxx-7d70145468f0",
+    "msg_timestamp": 1612764956322,
+    "nonce": "",
+    "verify_token": "xxx"
+  },
+  "sn": 9
+}"""
+
+        val ser = Signal.Event.serializer(SystemExtra.serializer())
+        val deserialized = json.decodeFromString(ser, rawJson)
+        val extra = deserialized.data.extra
+        assertIs<UpdateGuildEventExtra>(extra)
+        assertTrue(extra.body.enableOpen)
+    }
+
+    @Test
+    fun testDeserializeDeletedGuildEventWithBooleanEnableOpenFalse() {
+        //language=json
+        val rawJson =
+            """{
+  "s": 0,
+  "d": {
+    "channel_type": "GROUP",
+    "type": 255,
+    "target_id": "xxx",
+    "author_id": "1",
+    "content": "[系统消息]",
+    "extra": {
+      "type": "deleted_guild",
+      "body": {
+        "id": "xxx",
+        "name": "testDel",
+        "topic": "删除的服务器主题",
+        "user_id": "2418200000",
+        "icon": "",
+        "notify_type": 2,
+        "region": "beijing",
+        "enable_open": false,
+        "open_id": 0,
+        "default_channel_id": "xxxx",
+        "welcome_channel_id": "0"
+      }
+    },
+    "msg_id": "3d2bdb08-xxxx-faa2b9e77394",
+    "msg_timestamp": 1614086485182,
+    "nonce": "",
+    "verify_token": "xxx"
+  },
+  "sn": 210
+}"""
+
+        val ser = Signal.Event.serializer(SystemExtra.serializer())
+        val deserialized = json.decodeFromString(ser, rawJson)
+        val extra = deserialized.data.extra
+        assertIs<DeleteGuildEventExtra>(extra)
+        assertFalse(extra.body.enableOpen)
     }
 }

